### ISSUE TITLE
New BatchNorm divide by zero

### ISSUE THREAD

**Issue by ZagButNoZig** [patrick-kidger/equinox#1010]
Hey,

when using the new BatchNorm (`mode="batch"`) in inference mode with zero training steps, we do

```Python
scale = 1 - self.momentum**counter # 1 - x^0 = 0
mean = hidden_mean / scale
var = hidden_var / scale
```

This wasn't an issue before, as we would only update hidden during training.

**Comment by lockwo** [comment]
Ah good catch. I would just match the behavior of haiku (which I adapted it from) and flax and set the mean/var to the default inits (of 0/1)?

**Comment by ZagButNoZig** [comment]
Yeah, probably something like this would work:

```Python
scale = 1 - self.momentum**(jnp.where(counter == 0, 1, counter))
mean = hidden_mean / scale
var = hidden_var / scale
```
